qmarkdowntextedit.cpp

```diff
@@ -98,7 +98,9 @@ void QMarkdownTextEdit::setHighlightingEnabled(bool enabled) {
         return;
     }
 
-    _highlighter->setDocument(enabled ? document() : Q_NULLPTR);
+    if (_highlighter != nullptr) {
```

Comment:
Don't you better want to add that to line 97 `if (_highlightingEnabled == enabled)` and return in the beginning?

Reply:
It could also crash with `_highlighter->rehighlight();`...

Reply:
yes, fixed.

Reply:
> Don't you better want to add that to line 97 `if (_highlightingEnabled == enabled)` and return in the beginning?

I just did it that way. So i think it is the best and really simple